src/components/context/utilities.ts

```diff
@@ -167,6 +169,19 @@ export const getQuestionNameFromId = (
     .question;
 };
 
+export const getQuestionNames = (
+  responses: TableByUserResponse[],
+  questions: QuestionDataAppSetting[]
+) =>
+  responses
+    .map((res) => getQuestionNameFromId(questions, res.data.questionId))
+    .filter((r): r is string => Boolean(r))
+    .reduce((result, question) => {
+      const index = result.filter((entry) => entry[0] === question).length;
+      result.push([question, index]);
+      return result;
+    }, [] as [string, number][]);
```

Comment:
Her you can use the generic type on `.reduce<[string, number][]>(...)` to avoid the `never[]` error.